src/subsplit_dag.cpp

```diff
@@ -691,38 +701,52 @@ void SubsplitDAG::BuildDAGEdgesFromEdgeIndexer(BitsetSizeMap &edge_indexer) {
            "All edges should be bitsets with size 3 times taxon_count_.");
     const auto parent_id = GetDAGNodeId(edge.PCSPGetParentSubsplit());
     const auto child_id = GetDAGNodeId(edge.PCSPGetChildSubsplit());
-    SafeInsert(dag_edges_, {parent_id, child_id}, index);
+    auto left = GetDAGNode(parent_id).GetLeftLeafward();
+    auto clade = std::find(left.begin(), left.end(), child_id) != left.end()
```

Comment:
Should we assert that it appears in at least one of the clades? And would it be worth factoring this out into a function?

Reply:
I'm adding the assert. We already have a function that does that: FindNeighbor(). It returns `std::optional<std::tuple<LineId, Direction, Clade>>` which contains every possible information about the lookup: if it's found or not, it's edge id, direction and clade. We can't use FindNeighbor() here, because edges hasn't been added yet.